### ISSUE TITLE
sone search error

### ISSUE THREAD

**Issue by HieronymusCH** [Bombe/Sone#15]
Interner Fehler: Bitte melden Sie dies.

```
net.pterodactylus.util.template.TemplateException: Error while applying filter. (line 30, column 6)
	at net.pterodactylus.util.template.TemplateParser$Filters.filter(TemplateParser.java:646)
	at net.pterodactylus.util.template.FilteredPart.render(FilteredPart.java:63)
	at net.pterodactylus.util.template.ContainerPart.render(ContainerPart.java:78)
	at net.pterodactylus.util.template.ContainerPart.render(ContainerPart.java:78)
	at net.pterodactylus.util.template.Template.render(Template.java:27)
	at net.pterodactylus.util.template.TemplatePart.render(TemplatePart.java:72)
	at net.pterodactylus.util.template.ContainerPart.render(ContainerPart.java:78)
	at net.pterodactylus.util.template.LoopPart.render(LoopPart.java:124)
	at net.pterodactylus.util.template.ContainerPart.render(ContainerPart.java:78)
	at net.pterodactylus.util.template.ContainerPart.render(ContainerPart.java:78)
	at net.pterodactylus.util.template.Template.render(Template.java:27)
	at net.pterodactylus.sone.web.page.FreenetTemplatePage.handleRequest(FreenetTemplatePage.java:167)
	at net.pterodactylus.sone.web.page.FreenetTemplatePage.handleRequest(FreenetTemplatePage.java:53)
	at net.pterodactylus.sone.web.page.PageToadlet.handleRequest(PageToadlet.java:149)
	at net.pterodactylus.sone.web.page.PageToadlet.handleMethodGET(PageToadlet.java:107)
	at freenet.clients.http.ToadletContextImpl.callToadletMethod(ToadletContextImpl.java:721)
	at freenet.clients.http.ToadletContextImpl.handle(ToadletContextImpl.java:658)
	at freenet.clients.http.SimpleToadletServer$SocketHandler.run(SimpleToadletServer.java:1042)
	at freenet.support.PooledExecutor$MyThread.innerRun(PooledExecutor.java:248)
	at freenet.support.PooledExecutor$MyThread.realRun(PooledExecutor.java:188)
	at freenet.support.io.NativeThread.run(NativeThread.java:157)
Caused by: java.lang.IllegalStateException: FreenetURI(link).routingKey must not be null
	at net.pterodactylus.sone.text.SoneTextParser.toPart(SoneTextParser.kt:71)
	at net.pterodactylus.sone.text.SoneTextParser.access$toPart(SoneTextParser.kt:25)
	at net.pterodactylus.sone.text.SoneTextParser$splitLineIntoParts$1.invoke(SoneTextParser.kt:47)
	at net.pterodactylus.sone.text.SoneTextParser$splitLineIntoParts$1.invoke(SoneTextParser.kt:25)
	at kotlin.sequences.GeneratorSequence$iterator$1.calcNext(Sequences.kt:517)
	at kotlin.sequences.GeneratorSequence$iterator$1.hasNext(Sequences.kt:535)
	at kotlin.sequences.TransformingSequence$iterator$1.hasNext(Sequences.kt:152)
	at kotlin.sequences.SequencesKt___SequencesKt.toCollection(_Sequences.kt:633)
	at kotlin.sequences.SequencesKt___SequencesKt.toMutableList(_Sequences.kt:663)
	at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:654)
	at net.pterodactylus.sone.text.SoneTextParser.splitLineIntoParts(SoneTextParser.kt:51)
	at net.pterodactylus.sone.text.SoneTextParser.parse(SoneTextParser.kt:32)
	at net.pterodactylus.sone.template.ParserFilter.format(ParserFilter.kt:28)
	at net.pterodactylus.util.template.TemplateParser$Filters.filter(TemplateParser.java:644)
	... 20 more
```

I just started a search from the sone page.

Freenet 0.7.5 Build #1480 rbuild01480
USK@nwa8lHa271k2QvJ8aa0Ov7IHAV-DFOCFgmDt3X6BpCI,DuQSUZiI~agF8c-6tjsFFGuZ8eICrzWCILB60nT8KKo,AQACAAE/sone/78/sone-current.jar

**Comment by HieronymusCH** [comment]
This one happened after a click on /Sone/viewSone.html?sone=lGkNmLGAHZQFFvffh6YYPk8hDGzsdBGROTus66LmGqI

```
Interner Fehler: Bitte melden Sie dies.

net.pterodactylus.util.template.TemplateException: Error while applying filter. (line 55, column 27)
	at net.pterodactylus.util.template.TemplateParser$Filters.filter(TemplateParser.java:646)
	at net.pterodactylus.util.template.FilteredPart.render(FilteredPart.java:63)
	at net.pterodactylus.util.template.ContainerPart.render(ContainerPart.java:78)
	at net.pterodactylus.util.template.LoopPart.render(LoopPart.java:124)
	at net.pterodactylus.util.template.ContainerPart.render(ContainerPart.java:78)
	at net.pterodactylus.util.template.ConditionalPart.render(ConditionalPart.java:61)
	at net.pterodactylus.util.template.ContainerPart.render(ContainerPart.java:78)
	at net.pterodactylus.util.template.ContainerPart.render(ContainerPart.java:78)
	at net.pterodactylus.util.template.Template.render(Template.java:27)
	at net.pterodactylus.sone.web.page.FreenetTemplatePage.handleRequest(FreenetTemplatePage.java:167)
	at net.pterodactylus.sone.web.page.FreenetTemplatePage.handleRequest(FreenetTemplatePage.java:53)
	at net.pterodactylus.sone.web.page.PageToadlet.handleRequest(PageToadlet.java:149)
	at net.pterodactylus.sone.web.page.PageToadlet.handleMethodGET(PageToadlet.java:107)
	at freenet.clients.http.ToadletContextImpl.callToadletMethod(ToadletContextImpl.java:721)
	at freenet.clients.http.ToadletContextImpl.handle(ToadletContextImpl.java:658)
	at freenet.clients.http.SimpleToadletServer$SocketHandler.run(SimpleToadletServer.java:1042)
	at freenet.support.PooledExecutor$MyThread.innerRun(PooledExecutor.java:248)
	at freenet.support.PooledExecutor$MyThread.realRun(PooledExecutor.java:188)
	at freenet.support.io.NativeThread.run(NativeThread.java:157)
Caused by: java.lang.IllegalStateException: FreenetURI(link).docName must not be null
	at net.pterodactylus.sone.text.SoneTextParser.toPart(SoneTextParser.kt:71)
	at net.pterodactylus.sone.text.SoneTextParser.access$toPart(SoneTextParser.kt:25)
	at net.pterodactylus.sone.text.SoneTextParser$splitLineIntoParts$1.invoke(SoneTextParser.kt:47)
	at net.pterodactylus.sone.text.SoneTextParser$splitLineIntoParts$1.invoke(SoneTextParser.kt:25)
	at kotlin.sequences.GeneratorSequence$iterator$1.calcNext(Sequences.kt:517)
	at kotlin.sequences.GeneratorSequence$iterator$1.hasNext(Sequences.kt:535)
	at kotlin.sequences.TransformingSequence$iterator$1.hasNext(Sequences.kt:152)
	at kotlin.sequences.SequencesKt___SequencesKt.toCollection(_Sequences.kt:633)
	at kotlin.sequences.SequencesKt___SequencesKt.toMutableList(_Sequences.kt:663)
	at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:654)
	at net.pterodactylus.sone.text.SoneTextParser.splitLineIntoParts(SoneTextParser.kt:51)
	at net.pterodactylus.sone.text.SoneTextParser.parse(SoneTextParser.kt:32)
	at net.pterodactylus.sone.template.ParserFilter.format(ParserFilter.kt:28)
	at net.pterodactylus.util.template.TemplateParser$Filters.filter(TemplateParser.java:644)
	... 18 more

```

And by the way, are this reports even helpful?

**Comment by Bombe** [comment]
Oh, I fixed that bug two days ago!

**Comment by Bombe** [comment]
Fixed in a2369ab3b5b0f3f6dcc9d9fd1fcdc02c3003da64.
